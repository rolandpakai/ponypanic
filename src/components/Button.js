import {default as MuiButton} from '@mui/material/Button';
import LoadingButton from '@mui/lab/LoadingButton';
import CircularProgress from '@mui/material/CircularProgress';

const Button = (props) => {
  const { loading } = {...props};

  const buttonProps = {
    color: "error",
    size: "large",
    variant: "contained",
    fullWidth: true,
    sx: {
      borderRadius: 8,
      backgroundImage: "url(./images/button.svg)",
      backgroundSize: "cover",
      justifyContent: "center",
      alignItems: "center",
      fontSize: '1.25rem',
    },
    ...props
  }

  if(loading) {
    buttonProps.loadingIndicator = <CircularProgress color="inherit" size={25} sx={{color: '#fff'}}/>

    return ( 
      <LoadingButton 
        {...buttonProps}
      >
        {props.children}
      </LoadingButton>
    )
  } else {
    delete buttonProps.loading;
    
    return (
      <MuiButton 
        {...buttonProps}
      >
        {props.children}
      </MuiButton>   
    )
  } 
}

export default Button;