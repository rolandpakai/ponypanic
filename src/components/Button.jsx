import React from "react";
import LoadingButton from "@mui/lab/LoadingButton";
import CircularProgress from "@mui/material/CircularProgress";

const Button = (props) => {
  const { disabled, loading, onClick, children } = { ...props };

  const loadingIndicator = (
    <CircularProgress color="inherit" size={25} sx={{ color: "#fff" }} />
  );

  return (
    <LoadingButton
      disabled={disabled}
      loading={loading}
      onClick={onClick}
      color="error"
      size="large"
      variant="contained"
      fullWidth
      sx={{
        borderRadius: 8,
        backgroundImage: "url(./images/button.svg)",
        backgroundSize: "cover",
        justifyContent: "center",
        alignItems: "center",
        fontSize: "1.25rem",
      }}
      loadingIndicator={loadingIndicator}
    >
      {children}
    </LoadingButton>
  );
};

export default Button;
