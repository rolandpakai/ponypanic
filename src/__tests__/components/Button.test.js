import { render, act } from '@testing-library/react';
import Button from '../../components/Button';

describe('Button Component', () => {
  test('Button renders', () => {
    const {container} = render(<Button />);
    
		expect(container).toMatchSnapshot();
  });
});