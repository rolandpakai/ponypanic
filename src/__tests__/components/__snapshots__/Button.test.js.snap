// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button renders 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-ceotk2-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
